### PR TITLE
Fix error on out param of extern local function

### LINKED ISSUE #49500: Incorrect error about uninitialized out parameter in extern local function
**Version Used**: 
```
Branch master (18 Nov 2020)
Latest commit 823039e by David Wengier:
Create an unsafe method from a local function when necessary (#49389)
```

**Steps to Reproduce**:

```
using System.Runtime.InteropServices;

class C
{
  void M()
  {
    [DllImport("asdf")]
    static extern void Local(out int x); // error CS0177
  }
    
  [DllImport("asdf")]
  static extern void Method(out int x); // no errors
}
```

**Expected Behavior**:
No errors

**Actual Behavior**:
`error CS0177: The out parameter 'x' must be assigned to before control leaves the current method` is reported for `Local` but not for `Method`

Note that as the function is extern it isn't allowed to have a body so it's impossible to fix the error

### PR DESCRIPTION
**Pull request by RikkiGibson**
Closes #49500

### CODE REVIEW

File: src/Compilers/CSharp/Portable/FlowAnalysis/AbstractFlowPass_LocalFunctions.cs [hunk 1]
```diff
@@ -58,6 +58,12 @@ protected TLocalFunctionState GetOrCreateLocalFuncUsages(LocalFunctionSymbol loc
 
         public override BoundNode? VisitLocalFunctionStatement(BoundLocalFunctionStatement localFunc)
         {
+            if (localFunc.Symbol.IsExtern)
```
Comment (AlekseyTs) [inline src/Compilers/CSharp/Portable/FlowAnalysis/AbstractFlowPass_LocalFunctions.cs:61]: >localFunc.Symbol.IsExtern [](start = 16, length = 25)

Does this match the criteria used to skip analysis for type members (non-local functions)? Would it make sense to check for presence of the body instead?

**Review by AlekseyTs — APPROVED** [review]
LGTM (commit 2)

File: src/Compilers/CSharp/Portable/FlowAnalysis/AbstractFlowPass_LocalFunctions.cs (hunk 1 above)
Comment (RikkiGibson) [reply src/Compilers/CSharp/Portable/FlowAnalysis/AbstractFlowPass_LocalFunctions.cs:61]: I found that for member methods, we decide whether to analyze based on whether BindMethodBody returned a non-null body. `extern` member methods will force a `null` return from BindMethodBody, which makes us skip analysis even in an error scenario where an extern method has a body.

I didn't want to modify the conditions when a BoundLocalFunctionStatement is given a `null` body, in order to reduce churn, and felt that this check resulted in useful diagnostic behaviors (i.e. if the function is required to have a body but one is missing, we might as well also let you know about things that are required to be included in that body), which are consistent enough with the diagnostics given for member methods.

**Review by jcouv — APPROVED** [review]
LGTM Thanks (iteration 2)

**Pull request merged**